dtcw

```diff
@@ -6,84 +6,157 @@ set -e
+        install_component_and_exit "${environment}" "java"
     fi
-    export DTC_PROJECT_BRANCH
-fi
 
-# If DTC_HEADLESS provided (from tests), don't change it.
-if [ -z ${DTC_HEADLESS+x} ]; then
-    if [ -t 0 ]; then
-        DTC_HEADLESS=false
-    else
-        echo "Using headless mode since there is no (terminal) interaction possible"
-        DTC_HEADLESS=true
+    # No install command, so forward call to docToolchain but first we check if
+    # everything is there.
+    if [[ ${environment} != docker ]]; then
+        assert_doctoolchain_installed "${environment}" "${DTC_VERSION}"
+        assert_java_version_supported
+
+        # TODO: what if 'doctoolchain' found by $PATH does not match the one from the local environment?
+        # The version provided by $DTC_VERSION could be a different one.
     fi
-fi
-export DTC_HEADLESS
 
-DTC_OPTS="${DTC_OPTS:-} -PmainConfigFile=${MAIN_CONFIG_FILE} --warning-mode=none --no-daemon "
-
-echo "dtcw - docToolchain wrapper V0.38"
-echo "docToolchain V${DTC_VERSION}"
-
-# check if CLI, docker or sdkman are installed
-cli=false
-docker=false
-sdkman=false
-homefolder=false
-java=false
-doJavaCheck=true
-arch=$(uname -m)
-os=$(uname -s)
-
-if command -v doctoolchain &>/dev/null; then
-    echo "docToolchain as CLI available"
-    cli=true
-fi
-if command -v docker &>/dev/null; then
-    echo "docker available"
-    docker=true
-fi
-if command -v sdk &>/dev/null; then
-    echo "sdkman available"
-    sdkman=true
-fi
-if [ -d "${HOME}/.doctoolchain/docToolchain-${DTC_VERSION}/." ]; then
-    echo "home folder exists"
-    homefolder=true
-fi
-if [[ $# -lt 1 ]]; then
-    echo '
-Usage: ./dtcw [option...] [task...]
-
-You can use the same options and tasks as in underlying gradle.
-Use "./dtcw tasks --group doctoolchain" to see available tasks.
-Use "local", "sdk" or "docker" as first argument to force the use of a local, sdkman or docker install.
+    # TODO: can generateDeck, bakePreview be used in combination with other commands?
+    # The code below assumes we have just one task.
+
+    if [ "${1}" = "generateDeck" ]; then
+        # TODO: is this still true? The Dockerfile at
+        # https://github.com/docToolchain/docker-image/blob/ng-beta/alpine/Dockerfile
+        # seems to install reaveal.js also in the docker image.
+        assert_reveal_js_installed "${environment}"
+    fi
+
+    # If bakePreview is called, deactivate daemon.
+    if [ "${1}" = "bakePreview" ]; then
+        DTC_OPTS="${DTC_OPTS} -Dorg.gradle.daemon=false"
+    fi
+    # Don't shift command line arguments since the command is passed to
+    # 'doctoolchain'. This also means the check has to occur at the leatest time.
+
+    command=$(build_command "${environment}" "${DTC_VERSION}" "$@")
+
+    [[ "${DTC_HEADLESS}" = true ]] && echo "Using headless mode since there is no (terminal) interaction possible"
+
+    show_os_related_info
+
+    emu=""
+    if [ "$os" = "Darwin" ] && [ "$arch" = "arm64" ]; then
```

Comment:
I guess we run into a problem here with `install java`. 
The original idea behind the automatic use of the emulator was not to make the user think, but it will break the build:

The jdk will be installed for the arm architecture and here we switch to x86 architecture.
In addition, the goal is to switch to arm as soon as jbake work on arm.

what do you think?

Reply:
Well, this is the status quo what the current implementation is doing.
I would like to address the Java/ARM-Mac problem with #1082 and #1092.

I would not release the changes until we addressed a couple other open issues.

Reply:
I will address this topic in #1082.